sdk/python/feast/feature_store.py

```diff
@@ -22,13 +27,64 @@ class FeatureStore:
     """
 
     def __init__(
-        self, config_path: Optional[str], config: Optional[Config],
+        self, config_path: Optional[str] = None, config: Optional[Config] = None,
     ):
-        if config_path is None or config is None:
-            raise Exception("You cannot specify both config_path and config")
+        if config_path is not None and config is not None:
+            raise ValueError("You cannot specify both config_path and config")
         if config is not None:
             self.config = config
         elif config_path is not None:
             self.config = Config.from_path(config_path)
         else:
             self.config = Config()
+
+    def _pull_table(
```

Comment:
It seems like `_pull_table` would be better placed within the storage layer (`BigQueryStore`) itself, and not the `FeatureStore` class. I can't see this `feature_view.inputs.table_ref` holding up long term. Every data source will have their own implementation details. Like lets say we are pulling a parquet file that is compressed, where would the configuration for compression go?

I think the pull table argument should maybe take a `Union[DataSource, EntityDataFrame]`, and so the BigQueryOfflineStore should take `Union[BigQueryDataSource, BigQueryEntityDataFrame]` or something. At least we can fully retrieve the data from within the storage layer.

Reply:
Also, `_pull_table()` seems like it should be a function and not a method, so that it is easier to test.